flowmachine/flowmachine/features/subscriber/contact_balance.py

```diff
@@ -102,36 +106,14 @@ def __init__(
         self._cols = ["subscriber", "msisdn_counterpart", "events", "proportion"]
         super().__init__()
 
-    def _parse_tables_ensuring_direction_present(self, tables):
-
-        if isinstance(tables, str) and tables.lower() == "all":
-            tables = [f"events.{t}" for t in self.connection.subscriber_tables]
-        elif type(tables) is str:
-            tables = [tables]
-        else:
-            tables = tables
-
-        parsed_tables = []
-        tables_lacking_direction_column = []
-        for t in tables:
-            if "outgoing" in Table(t).column_names:
-                parsed_tables.append(t)
-            else:
-                tables_lacking_direction_column.append(t)
-
-        if tables_lacking_direction_column:
-            raise MissingDirectionColumnError(tables_lacking_direction_column)
-
-        return parsed_tables
-
     def _make_query(self):
 
         filters = []
         if self.direction != "both":
             filters.append(
-                f"outgoing IS {'TRUE' if self.direction == 'out' else 'FALSE'}"
+                f"outgoing = {'TRUE' if self.direction == 'out' else 'FALSE'}"
```

Comment:
FYI, the issue with the failing boolean clause due to the `WHERE ... IS TRUE` statement has been fixed in pglast (see [here](https://github.com/lelit/pglast/issues/11)). So we could revert this to your original version, but either is fine, it doesn't really matter.